azext_iot/iothub/providers/discovery.py

```diff
@@ -60,23 +67,27 @@ def get_policies(self, hub_name: str, rg: str) -> List:
         )
         policy_list = []
 
-        try:
-            while True:
-                policy_list.extend(policy_pager.advance_page())
-        except StopIteration:
-            pass
+        if ensure_min_version(TRACK_2_SDK_MIN_VERSION):
+            for policy in policy_pager.by_page():
+                policy_list.extend(policy)
+        else:
+            try:
+                while True:
+                    policy_list.extend(policy_pager.advance_page())
+            except StopIteration:
+                pass
 
         return policy_list
 
     def find_iothub(self, hub_name: str, rg: str = None):
         self._initialize_client()
 
-        from azure.mgmt.iothub.models import ErrorDetailsException
+        from azure.core.exceptions import HttpResponseError
 
         if rg:
             try:
                 return self.client.get(resource_group_name=rg, resource_name=hub_name)
-            except ErrorDetailsException:
+            except HttpResponseError:
```

Comment:
Will this exception work for both SDK types? And our error processing behaves the same?

Reply:
Btw python supports an exception tuple...for example `except (ErrorDetailsException, HttpResponseError):`. But one exception type would be preferred.

Reply:
Good catch, this is an issue I was working on yesterday - the `azure.core` exception import will cause an issue with the older CLI (2.3.1) - I'll have an update on this in a bit.

Reply:
So just to follow up, this issue is a bit complex.

- We switched from `ErrorDetailsException` to the `azure.core.HttpResponseError` (base error class) as `ErrorDetailsException` no longer exists in our Track 2 SDKs.
- In AZ CLI 2.3.1, however - we don't have the `azure.core` exceptions
- The actual exception this will return is a [ResourceNotFoundError](https://github.com/Azure/azure-sdk-for-python/blob/bfb7119dc7ea11aaf25d0f859f28c249b73312fd/sdk/core/azure-core/azure/core/exceptions.py#L317) which also comes from `azure.core`

Is it pertinent here to catch the specific error type if we're not utilizing the error/exception itself?
I'd like to avoid blanket `except:` statements (or more try/catch imports) but that might be the easiest fix here.

Reply:
Good details...my initial thought was if there was a common base exception like `Exception` we could use but I think the most pragmatic solution is the one you suggest in this specific case having `except:` since we are not using the exception itself.